javascript/javascript3/week1/weather.js

```diff
@@ -0,0 +1,126 @@
+const body = document.querySelector("body");
+const input = document.getElementById("input-city");
+const button = document.getElementById("button");
+
+button.addEventListener("click", getCityWeatherInfo);
+const apiKey = "aa5a826cc684a7127fa32b9ff1a5cf8e";
+
+//get data from API by city name
+function getCityWeatherInfo() {
+  let cityName = input.value;
+
+  if (!cityName) {
+    alert("please insert a city");
+  }
+
+  fetch(
+    `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${apiKey}`
+  )
+    .then((response) => response.json())
+    .then((data) => {
+      if (data.cod === "404") {
+        alert("It is not a city name");
+      }
+      renderWeatherInfo(data);
+    });
```

Comment:
To fix the local storage issue, you can update the local storage here when the user enters a city name. 

For example: 

```suggestion
    .then((data) => {
      if (data.cod === "404") {
        alert("It is not a city name");
      }
      localStorage.setItem("cityName", cityName);
      renderWeatherInfo(data);
    });
```


You can also check in the `useLocalStorage()` function if the item `cityName` in the local storage has a value, otherwise you can assign it a default value.

Reply:
Thank you so much for all comments, especially giphy limit and local storage. I learned much from them